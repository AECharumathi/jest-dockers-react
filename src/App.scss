@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Arima+Madurai:wght@200&family=Crimson+Text&family=Oswald&display=swap");

.App-header {
  background-color: #282c34;
  padding: 12px;
 // max-height: 5vh;
  display: flex;
  text-align: center;
  font-size:16px;
  color: #d6d6d6;
  font-family: 'Roboto';
  font-weight: 400;

  .header-container {
    width: 100%;
  }
  
}

body{
  //  min-height: 86vh;
    align-content: center;
}

.main-content {
//  min-height: 86vh;

  .main-content-container {
    width: 90%;
    margin: 0 auto;
  }

  .main-content-header {
    font-family: "Roboto";
    padding: 10px 10px;

    h2 {
      margin: 0px;
      font-size: 40px;
      text-decoration-line: underline; 
    }

    h6 {
        margin: 0px;
        font-size: 18px;
        color: #8f8d8d;
    }

  }

  .main-content-body {
    font-family: "Arima Madurai";
    padding: 0px 5px;
  }

  .main-content-quote {
    font-family: "Roboto";
    background-color: white;

    h2 {
      text-align: left;
      font-weight: 200;
    }

    h5{
        margin: 0px;
    }

  }

}

.display-flex{
  display:flex;
  justify-content: space-between;
}

.footer-container {
 max-height: 4vh;
  padding: 11px 80px;
  background-color: #282c34;
  font-family: 'Roboto';
  font-weight: 400;
  font-size: 18px;
  color: rgb(117, 115, 115);
  display: flex !important;
  justify-content: space-between;
}


