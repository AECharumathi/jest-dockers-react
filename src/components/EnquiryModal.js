import React, { Component } from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import {
  Form
} from "react-bootstrap";

import { connect } from 'react-redux';

import { postEnquiry } from "../action/index";
import "../components/styles.scss";

class EnquiryModal extends Component {

  constructor(props) {
    super(props);
    console.log("PROPS " + JSON.stringify(this.props));
    this.state = {
      firstName: '',
      lastName: '',
      occupation: '',
      companyName: '',
      emailId: '',
      emailError: '',
      phoneNumber: '',
      phoneError: '',
      viewForm: true
    }
    this.closeUserForm = this.closeUserForm.bind(this)

  }

  closeUserForm = () => {
    this.setState({ viewForm: false })
    this.props.closeForm();
  }

  getCourseName = () => {
    return this.props.courseListId;
  }

  handlePhoneNumber = (e) => {
    this.setState({
      phoneNumber: e.target.value,
      phoneError: e.target.validationMessage
    })
  }
  addUser = () => {
    let cName = this.getCourseName();
    let userDetails = {
      "firstName": this.state.firstName,
      "lastName": this.state.lastName,
      "occupation": this.state.occupation,
      "companyName": this.state.companyName,
      "emailId": this.state.emailId,
      "phoneNumber": this.state.phoneNumber,
      "courseName": cName
    }
    postEnquiry(userDetails, this.onSuccess);
  }

  onSuccess = () => {
    this.closeUserForm();
  }

  componentDidMount() {
    console.log("course name " + this.props.courseListId);
  }



  render() {
    return (
      <>

        <Modal isOpen={this.state.viewForm} toggle={this.closeUserForm} className={this.className} id="modal" >
          <ModalHeader toggle={this.closeUserForm}><h2>Course Enquiry</h2></ModalHeader>
          <ModalBody>
            <Form>
              <Form.Group controlId="formFirstName">
                <Form.Label>
                  First Name<span class="required">*</span>
                </Form.Label>
                <Form.Control
                  type="firstName"
                  value={this.state.firstName}
                  // ref={firstNameRef}
                  onChange={(e) => this.setState({ firstName: e.target.value })}
                  maxLength="25"
                  autoComplete="off"
                />
              </Form.Group>
              <Form.Group controlId="formLastName">
                <Form.Label>
                  Last Name<span class="required">*</span>
                </Form.Label>
                <Form.Control
                  type="LastName"
                  value={this.state.lastName}
                  // ref={firstNameRef}
                  onChange={(e) => this.setState({ lastName: e.target.value })}
                  maxLength="25"
                  autoComplete="off"
                />
              </Form.Group>
              <Form.Group controlId="formOccupation ">
                <Form.Label>
                  Occupation  <span class="required">*</span>
                </Form.Label>
                <Form.Control
                  as="select"
                  value={this.state.occupation}
                  onChange={(e) => this.setState({ occupation: e.target.value })}
                >
                  <option>Select</option>
                  <option value="IT Professional">IT Professional</option>
                  <option value="Vendor">Vendor</option>
                  <option value="student">
                    Student
                  </option>
                  <option value="Others">Others</option>
                </Form.Control>
              </Form.Group>

              <Form.Group controlId="formCompanyName">
                <Form.Label>
                  Organization<span class="required">*</span>
                </Form.Label>
                <Form.Control
                  type="companyName"
                  value={this.state.companyName}
                  onChange={(e) => this.setState({ companyName: e.target.value })}
                  maxLength="25"
                  autoComplete="off"
                />
              </Form.Group>

              <Form.Group controlId="formEmailId">
                <Form.Label>
                  Email Id<span class="required">*</span>
                </Form.Label>
                <span className="invalid-feedback d-block">
                  {this.state.emailError}
                </span>
                <Form.Control
                  type="emailId"
                  value={this.state.emailId}
                  onChange={(e) => this.setState({ emailId: e.target.value })}
                  maxLength="250"
                  autoComplete="off"
                  pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2, 4}$"
                />
              </Form.Group>
              <Form.Group controlId="formPhoneNumber">
                <Form.Label>
                  Phone Number<span class="required">*</span>
                </Form.Label>
                <span className="invalid-feedback d-block">
                  {this.state.phoneError}
                </span>
                <Form.Control
                  type="phoneNumber"
                  value={this.state.phoneNumber}
                  // ref={firstNameRef}
                  onChange={this.handlePhoneNumber}
                  maxLength="10"
                  autoComplete="off"
                  // pattern="[A-Za-z]{3}"
                  pattern="^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$"
                />

              </Form.Group>
            </Form>
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={this.addUser}>Enquire</Button>{' '}
            <Button color="secondary" onClick={this.closeUserForm}>Cancel</Button>
          </ModalFooter>
        </Modal>

      </>
    );
  }
}


// function mapDispatchToProps(dispatch){
//   return bindActionCreators({postEnquiry}, dispatch)
// }

export default connect()(EnquiryModal)