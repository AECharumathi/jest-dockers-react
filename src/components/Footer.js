import React, { Component } from 'react';

export default class Footer extends Component {

    render() {
        return (
            <>
                <div className="footer-container">
                    <span>Terms and Conditions</span>
                    <span>FAQ</span>
                    <span>Contact Us</span>
                </div>

            </>
        );
    }
}