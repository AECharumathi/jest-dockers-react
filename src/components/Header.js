import React, { Component } from 'react';

export default class Header extends Component {

    render() {
        return (
            <>
                <div className="header-container">
                    <h1>COURSE ENQUIRY</h1>
                </div>

            </>
        );
    }
}