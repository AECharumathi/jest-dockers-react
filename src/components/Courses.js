import React, { Component } from 'react';
import { Link } from "react-router-dom";
import { connect } from 'react-redux';

import "../components/styles.scss";
import EnquiryModal from './EnquiryModal';

import { bindActionCreators } from 'redux';

import { getCourses } from "../action";

class Courses extends Component {

    constructor(props) {
        super(props);
        this.state = {
            courseList: [],
            showUserForm: false,
            firstName: '',
            lastName: '',
            emailId: '',
            phoneNumber: '',
            nationality: '',
            zipCode: '',
            courseName: '',
            className: props.className
        }
        this.displayUserForm = this.displayUserForm.bind(this);
        this.closeUserForm = this.closeUserForm.bind(this);
    }

    componentDidMount() {
        this.props.getCourses();
        console.log("inside componentDidMount")
    }

    componentDidUpdate(prevProps) {
        if (this.state.courseList.length !== this.props.courses.courses.data.length)
            this.setState({
                courseList: this.props.courses.courses.data
            })
    }

    displayUserForm = (name) => {
        this.setState({
            courseName: name,
            showUserForm: true
        })
    }

    closeUserForm = () => {
        this.setState({
            showUserForm: false
        })
    }

    render() {
        return (
            <>
                <div className="main-content">
                    <div className={`main-content-container ${this.state.showUserForm ? "style-opacity" : ""}`}>
                        <div className="main-content-header display-flex">
                            <h2>List of Courses</h2>
                            <div className="button-container">
                                <button className="style-btn"><Link to="/userEnquired" >Users Enquired</Link></button>
                                <button className="style-btn"><Link to="/userForum" >Forum</Link></button>
                            </div>
                        </div>
                        <div className="main-content-body">
                            <div className="main-content-list">
                                {this.state.courseList.length > 0 && this.state.courseList.map((course, index) => {
                                    return <div className="course-card">
                                        <span className="course-name">{course.courseName}</span>
                                        <p>{course.courseSortDesc}</p>
                                        <button className="enquire-button" onClick={() => this.displayUserForm(course.courseName)}>Enquire</button>

                                    </div>
                                })
                                }
                            </div>

                        </div>
                        {this.state.showUserForm ? <EnquiryModal closeForm={this.closeUserForm} courseListId={this.state.courseName} /> : null
                        }
                    </div>
                </div>

            </>
        );
    }
}

function mapStateToProps(state) {
    return {
        courses: state.courses
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ getCourses }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Courses)