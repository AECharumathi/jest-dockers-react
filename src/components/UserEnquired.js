import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from "react-router-dom";
import { bindActionCreators } from 'redux';
import { Table } from "react-bootstrap";

import goBack from "../asset/chevron-left-solid.svg";
import "../components/styles.scss";
import "../App.scss";

import { getUsers } from "../action";

class UserEnquired extends Component {

    constructor(props) {
        super(props);
        this.state = {
            userList: [],
            courseList: []
        }
    }

    componentDidMount() {
        console.log("userEnquired");
        this.props.getUsers();
    }

    componentDidUpdate(prevProps) {
        if (this.state.userList.length !== this.props.users.users.data.length)
            this.setState({
                userList: this.props.users.users.data
            })
    }

    render() {
        return (
            <>
                <div className="main-content">
                    <div className="main-content-container">

                        <div className="main-content-header main-content-list">
                            <Link to="/courses" className="style-goBack-container" ><img src={goBack} alt="export" className="style-goBack-img" /> </Link>
                            <h2 className="main-content-header">Users Enquired</h2>
                        </div>
                        <div className="main-content-body">
                            <div className="main-content-quote">
                                <Table className="table-main">
                                    <thead className="table-caption">
                                        <tr>
                                            <th>Id</th>
                                            <th > First Name </th>
                                            <th > Last Name </th>
                                            <th  > Email Id </th>
                                            <th >Phone Number</th>
                                            <th >Occupation</th>
                                            <th >Company Name</th>
                                            <th >Course Enquired</th>
                                        </tr>
                                    </thead>
                                    <tbody className="table-body">
                                        {this.state.userList.length > 0 ?
                                            this.state.userList.map((user, index) => {
                                                return <tr>
                                                    <td >{user.id}</td>
                                                    <td > {user.firstName} </td>
                                                    <td > {user.lastName} </td>
                                                    <td > {user.emailId} </td>
                                                    <td  > {user.phoneNumber} </td>
                                                    <td  > {user.occupation} </td>
                                                    <td > {user.companyName} </td>
                                                    <td > {user.courseName}</td>
                                                </tr>
                                            }) : null

                                        }
                                    </tbody>
                                </Table>
                            </div>

                        </div>
                    </div>


                </div>

            </>
        );
    }
}

function mapStateToProps(state) {
    return {
        users: state.users
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({ getUsers }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(UserEnquired)