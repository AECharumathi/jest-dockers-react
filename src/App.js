import './App.scss';

import {
  BrowserRouter as Router,
  Redirect,
  Route
} from "react-router-dom";

import Header from './components/Header';
import Courses from "./components/Courses";
import UserEnquired from "./components/UserEnquired";
import UserForum from "./components/UserForum";
import Footer from "./components/Footer";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <Header />
      </header>
      <body>
        <Router>
          <Route path="/courses" children={<Courses />} />
          <Route path="/userEnquired" children={<UserEnquired />} />
          <Route path="/userForum" children={<UserForum />} />
          <Redirect from="/" to="/courses" />
        </Router>
      </body>
      <footer className="App-footer">
        <Footer />
      </footer>
    </div>
  );
}

export default App;
