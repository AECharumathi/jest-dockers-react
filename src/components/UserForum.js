import React, { Component } from 'react';

import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import { Link } from "react-router-dom";

import goBack from "../asset/chevron-left-solid.svg";
import { postComment, getComment } from "../action/index";
import "../components/styles.scss";
import "../App.scss";


class UserForum extends Component {

  constructor(props) {
    super(props);

    this.state = {
      userName: '',
      comment: '',
      commentError: '',
      listComment: [],
      viewForm: true
    }
    this.closeUserForm = this.closeUserForm.bind(this)

  }

  closeUserForm = () => {
    this.setState({ viewForm: false })
    this.props.closeForm();
  }

  handleUserName = (e) => {
    this.setState({ userName: e.target.value })
  }

  handleComment = (e) => {
    this.setState({ comment: e.target.value })
  }

  addUserComment = () => {

    var today = new Date();
    var date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate();
    var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
    var dateTime = date + ' ' + time;

    let commentDetails = {
      "userName": this.state.userName,
      "comment": this.state.comment,
      "commentTime": dateTime
    }
    postComment(commentDetails, this.onSuccessComment);
    this.setState({ userName:"", comment: "" })
    }

  componentDidMount() {
    this.props.getComment();
  }

  componentDidUpdate(prevProps) {

    if (this.state.listComment.length !== this.props.commentList.commentList.data.length) {

      this.setState({
        listComment: this.props.commentList.commentList.data
      })
    }
  }

  onSuccessComment = () => {
    alert("Comment successfully submitted");
    this.props.getComment();
  }
  render() {
    return (
      <>
        <div className=" main-content">
          <div className="main-content-container">
            <div className="main-content-header main-content-list">
            <Link to="/courses" className="style-goBack-container" ><img src={goBack} alt="export" className="style-goBack-img" /> </Link>
              <h2 className="main-content-header">User Forum</h2>
            </div>
            <div className="forum-container main-content-body">
              <div className="justify-content-between">
                <h2>User Queries</h2>
                {this.state.listComment.length > 0 && this.state.listComment.map((item, index) => {
                  return <div className="style-user-comment-container" >
                    <div className="name-container">
                      <span className="bold-feild style-comment-user" >{item.userName} | {item.commentTime}</span>
                    </div>
                    <div className="name-container">
                      <span className="" >{item.comment}</span>
                    </div>
                  </div>
                })}
              </div>

              <div className="justify-content-between style-height">
                <span style={{ color: "red", fontFamily: "Roboto condensed" }} >
                  {this.state.userError}
                </span>
                <div >
                  <h2>New Comment</h2>
                  <div className="name-container">
                    <label>User name</label>
                    <input onChange={this.handleUserName} placeholder="Enter your name" value={this.state.userName} maxLength="50" className="style-input-feild" />
                  </div>
                  <div className="name-container">
                    <label>Comment</label>
                    <textarea onChange={this.handleComment} value={this.state.comment} maxLength="2000" className="style-textArea" >
                    </textarea>
                  </div>
                </div>
                <div className="name-container">
                  <button className="enquire-button" onClick={() => this.addUserComment()}>Add Comment</button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </>
    );
  }
}


function mapStateToProps(state) {
  return {
    commentList: state.commentList
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ postComment, getComment }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(UserForum)
