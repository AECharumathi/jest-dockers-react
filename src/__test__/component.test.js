import React from 'react';
import { create } from 'react-test-renderer';

import App from "../App";
import Courses from "../components/Courses";
import EnquiryModal from "../components/EnquiryModal";
import Header from "../components/Header";
import Footer from "../components/Footer";
import UserEnquired from "../components/UserEnquired";
import UserForum from "../components/UserForum";


describe('App Component Snapshot', ()=>{
    test("App", ()=>{
        let appComponent = create(<App/>);
        expect(appComponent.toJSON()).toMatchSnapshot();
    })
})

describe('Header Component Snapshot', ()=>{
    test("Header", ()=>{
        let header = create(<Header/>);
        expect(header.toJSON()).toMatchSnapshot();
    })
})

// describe('Courses Component Snapshot', ()=>{
//     test("Courses", ()=>{
//         let courses = create(<Courses/>);
//         expect(courses.toJSON()).toMatchSnapshot();
//     })
// })

// describe('EnquiryModal Component Snapshot', ()=>{
//     test("EnquiryModal", ()=>{
//         let enquiryModal = create(<EnquiryModal/>);
//         expect(enquiryModal.toJSON()).toMatchSnapshot();
//     })
// })


// describe('UserEnquired Component Snapshot', ()=>{
//     test("UserEnquired", ()=>{
//         let userEnquired = create(<UserEnquired/>);
//         expect(userEnquired.toJSON()).toMatchSnapshot();
//     })
// })

// describe('UserForum Component Snapshot', ()=>{
//     test("UserForum", ()=>{
//         let userForum = create(<UserForum/>);
//         expect(userForum.toJSON()).toMatchSnapshot();
//     })
// })

describe('Footer Component Snapshot', ()=>{
    test("Footer", ()=>{
        let footer = create(<Footer/>);
        expect(footer.toJSON()).toMatchSnapshot();
    })
})