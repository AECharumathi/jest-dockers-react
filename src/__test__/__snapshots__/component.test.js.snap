// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot App 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="header-container"
    >
      <h1>
        COURSE ENQUIRY
      </h1>
    </div>
  </header>
  <body />
  <footer
    className="App-footer"
  >
    <div
      className="footer-container"
    >
      <span>
        Terms and Conditions
      </span>
      <span>
        FAQ
      </span>
      <span>
        Contact Us
      </span>
    </div>
  </footer>
</div>
`;

exports[`Footer Component Snapshot Footer 1`] = `
<div
  className="footer-container"
>
  <span>
    Terms and Conditions
  </span>
  <span>
    FAQ
  </span>
  <span>
    Contact Us
  </span>
</div>
`;

exports[`Header Component Snapshot Header 1`] = `
<div
  className="header-container"
>
  <h1>
    COURSE ENQUIRY
  </h1>
</div>
`;
